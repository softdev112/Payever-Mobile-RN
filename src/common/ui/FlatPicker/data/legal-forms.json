{
  "data": [
    {
      "value": "",
      "label": "Legal Form"
    },
    {
      "value": "LEGAL_FORM_AG",
      "label": "Public Limited Company"
    },
    {
      "value": "LEGAL_FORM_EINZELUNTERN",
      "label": "Individual Enterprise"
    },
    {
      "value": "LEGAL_FORM_GBR",
      "label": "Civil Law Partnership"
    },
    {
      "value": "LEGAL_FORM_GMBH",
      "label": "Limited Liability Company"
    },
    {
      "value": "LEGAL_FORM_KG",
      "label": "Limited Partnership"
    },
    {
      "value": "LEGAL_FORM_OHG",
      "label": "General Partnership"
    },
    {
      "value": "LEGAL_FORM_SONSTIGES",
      "label": "Other"
    },
    {
      "value": "LEGAL_FORM_UG",
      "label": "Limited"
    },
    {
      "value": "LEGAL_FORM_EK",
      "label": "Sole Trader"
    },
    {
      "value": "LEGAL_FORM_EV",
      "label": "Registered Association"
    },
    {
      "value": "LEGAL_FORM_EG",
      "label": "Registered Cooperative"
    }
  ]
}