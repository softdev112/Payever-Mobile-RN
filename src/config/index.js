import generatedConfig from './_autogenerated';

export class Config {
  siteUrl: string;
  api: {
    baseUrl: string;
    clientId: string;
    clientSecret: string;
  };
  debug: {
    consoleLevel: boolean;
    logApiCall: boolean;
    logMobx: boolean;
    logSLevel: boolean;
    logSUrl: string;
    logWampCall: boolean;
  };

  constructor(data) {
    data.debug = data.debug || {};
    Object.assign(this, data);
  }
}

export default new Config(generatedConfig);