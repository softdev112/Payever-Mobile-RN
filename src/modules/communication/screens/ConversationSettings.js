import { Component } from 'react';
import { inject, observer } from 'mobx-react/native';
import { ScrollView, TouchableOpacity } from 'react-native';
import { Navigator } from 'react-native-navigation';
import { ErrorBox, Loader, NavBar, StyleSheet, Text, View } from 'ui';

import type CommunicationStore from '../../../store/communication';
import Avatar from '../components/contacts/Avatar';
import CheckBoxPref from '../components/settings/CheckBoxPref';

@inject('communication')
@observer
export default class ConversationSettings extends Component {
  static navigatorStyle = {
    navBarHidden: true,
  };

  props: {
    communication: CommunicationStore;
    navigator: Navigator;
  };

  componentWillMount() {
    const { communication } = this.props;
    communication.getConversationSettings();
  }

  onConvNotificationPropChange(value) {
    this.props.communication.changeConvNotificationProp(value);
  }

  onOfferLinkPress(offerId: number) {
    this.props.navigator.showModal({
      screen: 'marketing.ViewOffer',
      animated: true,
      passProps: {
        offerId,
      },
    });
  }

  renderOffers(offers: OfferRow) {
    return offers.map((offer) => {
      return (
        <TouchableOpacity
          style={styles.offerBtn}
          key={offer.id}
          onPress={() => this.onOfferLinkPress(offer.id)}
        >
          <Text style={styles.offerText}>{offer.created_at} #{offer.id}</Text>
        </TouchableOpacity>
      );
    });
  }

  render() {
    const {
      isLoading,
      error,
      selectedConversationSettings,
    } = this.props.communication;

    return (
      <View style={styles.container}>
        <NavBar popup>
          <NavBar.Back />
          <NavBar.Title title="Conversation Settings" />
        </NavBar>
        <Loader isLoading={isLoading}>
          {error || !selectedConversationSettings ? (
            <ErrorBox message={error} />
          ) : (
            <View style={styles.userInfo}>
              <Avatar
                style={styles.avatar}
                lettersStyle={styles.avatarLetters}
                avatar={selectedConversationSettings.avatar}
              />
              <Text style={styles.name}>
                {selectedConversationSettings.name}
              </Text>
              <CheckBoxPref
                initValue={selectedConversationSettings.notification}
                title="Notifications"
                icon="fa-bell-o"
                onValueChange={::this.onConvNotificationPropChange}
              />
              {selectedConversationSettings.offers.length > 0 && (
                <View style={styles.offersContainer}>
                  <Text style={styles.sentOffersText}>Sent Offers:</Text>
                  <ScrollView showsVerticalScrollIndicator={false}>
                    {this.renderOffers(selectedConversationSettings.offers)}
                  </ScrollView>
                </View>
              )}
            </View>
          )}
        </Loader>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  userInfo: {
    flex: 1,
    paddingTop: 20,
    paddingBottom: 8,
    paddingHorizontal: 15,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },

  avatar: {
    width: 70,
    height: 70,
    marginBottom: 10,
    marginRight: 0,
  },

  avatarLetters: {
    fontSize: 34,
  },

  name: {
    fontSize: 24,
    fontWeight: '300',
    marginBottom: 15,
  },

  sentOffersText: {
    alignSelf: 'flex-start',
    fontSize: 18,
    marginBottom: 4,
  },

  offersContainer: {
    flex: 1,
    alignSelf: 'stretch',
    paddingHorizontal: 8,
  },

  offerBtn: {
    padding: 3,
  },

  offerText: {
    color: '$pe_color_blue',
  },
});

type OfferRow = {
  id: number;
  created_at: string;
  slug: string;
};