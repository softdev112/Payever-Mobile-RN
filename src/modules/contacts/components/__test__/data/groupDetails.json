{
  "id": 26,
  "name": "GOSU",
  "filters": null,
  "contact_models": [
    {
      "id": 48011,
      "first_name": "Dart",
      "last_name": "Dart",
      "email": "sdasd@df.yu",
      "country": "US",
      "city": "fdsfdsfsd",
      "zip_code": "12312321",
      "street": "dsfdsfsdf",
      "phone": "213123123",
      "avatar_url": null,
      "count_purchases": 0,
      "birthday": "2016-05-24T00:00:00+00:00",
      "country_name": "United States",
      "invitation_date": "2017-05-24 14:20",
      "invitation_available": true,
      "registered": null,
      "business_name": null,
      "business_logo_url": null,
      "business_slug": null
    },
    {
      "id": 48016,
      "first_name": "Droideck",
      "last_name": "1",
      "email": "asdas@dsa.ty",
      "country": "US",
      "city": "xczxcz",
      "zip_code": "12312",
      "street": "fsdfsdf",
      "phone": "123423423",
      "avatar_url": "https://stage.payever.de/media/cache/contact_avatar.square.64.retina/cc/9a/cc9a7db0cfc109e09583aca4117f2a62.jpg",
      "count_purchases": 0,
      "birthday": null,
      "country_name": "United States",
      "invitation_date": "2017-05-24 14:39",
      "invitation_available": true,
      "registered": null,
      "business_name": null,
      "business_logo_url": null,
      "business_slug": null
    },
    {
      "id": 48017,
      "first_name": "Droideck 2",
      "last_name": "Droideck 1",
      "email": "asdas@rt.rt",
      "country": "US",
      "city": "sdfdsfsd",
      "zip_code": "123123",
      "street": "sadasdas",
      "phone": "12312312",
      "avatar_url": "https://stage.payever.de/media/cache/contact_avatar.square.64.retina/1b/5d/1b5d3514001b34dfd64d5351433f6461.jpg",
      "count_purchases": 0,
      "birthday": "2018-06-24T00:00:00+00:00",
      "country_name": "United States",
      "invitation_date": "2017-05-24 14:48",
      "invitation_available": true,
      "registered": null,
      "business_name": null,
      "business_logo_url": null,
      "business_slug": null
    },
    {
      "id": 48018,
      "first_name": "Anakine",
      "last_name": "SkyWalker",
      "email": "asdas@dsf.rt",
      "country": "US",
      "city": "dsfsdfsd",
      "zip_code": "12312312",
      "street": "dsfdsfdsfs",
      "phone": "12312312",
      "avatar_url": "https://stage.payever.de/media/cache/contact_avatar.square.64.retina/13/7a/137af7a32842bc226863acb890e58d11.jpg",
      "count_purchases": 0,
      "birthday": "2018-06-24T00:00:00+00:00",
      "country_name": "United States",
      "invitation_date": "2017-05-24 14:51",
      "invitation_available": true,
      "registered": null,
      "business_name": null,
      "business_logo_url": null,
      "business_slug": null
    },
    { "id": 48019,
      "first_name": "Anakine1",
      "last_name": "SkyWalker",
      "email": "asdas@ds232f.rt",
      "country": "US",
      "city": "dsfsdfsd",
      "zip_code": "12312312",
      "street": "dsfdsfdsfs",
      "phone": "12312312",
      "avatar_url": "https://stage.payever.de/media/cache/contact_avatar.square.64.retina/58/05/58055b3d4ccec9462b991f0094655c5b.jpg",
      "count_purchases": 0,
      "birthday": null,
      "country_name": "United States",
      "invitation_date": "2017-05-26 07:45",
      "invitation_available": true,
      "registered": null,
      "business_name": null,
      "business_logo_url": null,
      "business_slug": null
    }
  ],
  "logo_url": null,
  "total_spent": 0,
  "count_purchases": 0,
  "orders_count": 0
}