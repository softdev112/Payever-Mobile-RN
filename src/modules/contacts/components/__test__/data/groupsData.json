{
  "pagination_data": {
    "last": 1,
    "current": "1",
    "numItemsPerPage": 20,
    "first": 1,
    "pageCount": 1,
    "totalCount": 4,
    "pageRange": 1,
    "startPage": 1,
    "endPage": 1,
    "pagesInRange": [ 1 ],
    "firstPageInRange": 1,
    "lastPageInRange": 1,
    "currentItemCount": 4,
    "firstItemNumber": 1,
    "lastItemNumber": 4
  },

  "group_list":
    [
      {
        "id": 24,
        "name": "fsdfsdfsd",
        "contacts_count": 0,
        "logo_url": "https://stage.payever.de/media/cache/group_logo.square.64.retina/b9/ae/b9ae0739ed595b8dc551324fd5318255.jpg",
        "total_spent": "0.00",
        "count_purchases": "0",
        "orders_count": "0"
      },
      {
        "id": 26,
        "name": "GOSU",
        "contacts_count": 5,
        "logo_url": null,
        "total_spent": "0.00",
        "count_purchases": "0",
        "orders_count": "0"
      },
      {
        "id": 46,
        "name": "Grou",
        "contacts_count": 0,
        "logo_url": "https://stage.payever.de/media/cache/group_logo.square.64.retina/58/09/5809ae0c23b9069e717b9b7c35dc5249.jpg",
        "total_spent": "0.00",
        "count_purchases": "0",
        "orders_count": "0"
      },
      {
        "id": 45,
        "name": "Group2410",
        "contacts_count": 0,
        "logo_url": "https://stage.payever.de/media/cache/group_logo.square.64.retina/a0/70/a070397f32a92a71385b56d4329a26bd.jpg",
        "total_spent": "0.00",
        "count_purchases": "0",
        "orders_count": "0"
      }
    ]
}
