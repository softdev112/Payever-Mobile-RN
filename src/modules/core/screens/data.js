/* eslint-disable */
export default {
  since: 19701,
  id: 1511,
  status: 'SENT',
  type: 'EMAIL',
  visibility: 'ALL',
  marketing_channel_set: {
    id: 75515,
    name: 'Personal offer from MyOnlineShop',
    slug: 'ninjateampierreduergen-macsupport-itservices-designpassarinho-cachaca-tradingpassarinho-cachaca-tradingglobal-sales-incglobal-sales-incmy-online-shop/marketing/personal-offer-from-myonlineshop-28',
    store: {
      id: 12179,
      business: {
        name: 'Media Markt',
        slug: 'my-online-shop',
        logo: 'https://mein.payever.de/media/cache/business_logo.big.uncropped/1c/bf/1cbf2a72f21eac5a8677936d4d7a7d98.png',
      },
      items: [
        {
          id: 1923,
          item_type: 'physical',
          name: 'iMac',
          description: 'With Retina display, text appears sharper than ever, videos are unbelievably lifelike and you can see new levels of detail in your photos. The 21.5-inch iMac is available with the all-new Retina 4K display featuring an amazing 4096 x 2304 resolution and 9.4 million pixels, 4.5 times more than the standard 21.5-inch iMac display.  Every 27-inch iMac features a Retina 5K display, the worldâ€™s highest resolution all-in-one display with a stunning 14.7 million pixels, 7 times more pixels than an HD display.',
          positions: [
            {
              id: 4311,
              name: '27-inch with Retina 5K display',
              description: null,
              price: 1799,
              quantity: 1,
              barcode: null
            }
          ],
          media: [
            {
              id: 2385,
              thumbnail: 'https://mein.payever.de/media/cache/item_list/d9/15/d91551911fbd4ba1578d634b488ea9b6.jpg',
              is_video: false
            }
          ],
          like_profiles: [],
          item_shipping_type: 'general',
          fixed_shipping_price: null,
          weight: null,
          vat: '100',
          origin_image: 'https://mein.payever.de/media/cache/resolve/customer_shop_item.origin/d9/15/d91551911fbd4ba1578d634b488ea9b6.jpg',
          thumbnail: 'https://mein.payever.de/media/cache/customer_shop_item/d9/15/d91551911fbd4ba1578d634b488ea9b6.jpg',
          thumbnailList: 'https://mein.payever.de/media/cache/customer_shop_item.list/d9/15/d91551911fbd4ba1578d634b488ea9b6.jpg',
          thumbnailGrid: 'https://mein.payever.de/media/cache/customer_shop_item.grid/d9/15/d91551911fbd4ba1578d634b488ea9b6.jpg',
          market_url: 'https://mein.payever.de/market/#product/1923',
        }
      ],
      logo_url: null,
    },
  },
  title: 'Personal offer from MyOnlineShop',
  description: '<h1>A special offer. Just for you.</h1><p>As you previously bought from us we wanted to thank you by showing you this offer below before everyone else sees it.</p><p>Best,<br/>MyOnlineShop</p>',
  created_at: '2017-02-13T08:54:32+01:00'
};