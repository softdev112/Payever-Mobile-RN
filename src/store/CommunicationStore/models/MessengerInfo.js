import { computed, extendObservable, observable } from 'mobx';

import ConversationInfo from './ConversationInfo';
import UserSettings from './UserSettings';
import Group from './Group';
import type { MessengerData } from '../../../common/api/MessengerApi';
import Avatar from './Avatar';

export default class MessengerInfo {
  @observable conversations: Array<ConversationInfo> = [];
  @observable groups: Array<Group>                   = [];
  @observable marketingGroups: Array<Group>          = [];
  @observable messengerUser: {
    avatar: Avatar;
    id: number;
    name: string;
    user_type: 'business' | 'user';
  };
  @observable userSettings: UserSettings;
  @observable wsUrl: string;

  constructor(data: MessengerData) {
    data.conversations = (data.conversations || [])
      .map(c => new ConversationInfo(c));

    data.groups = (data.groups || [])
      .map(g => new Group(g));

    data.marketingGroups = data.marketingGroups || [];

    data.messengerUser = data.messengerUser || {};
    data.messengerUser.avatar = new Avatar(data.messengerUser.avatar);

    data.userSettings = new UserSettings(data.userSettings);

    extendObservable(this, data);
  }

  @computed get isBusiness() {
    return this.messengerUser.user_type === 'business';
  }

  @computed
  get unreadCount() {
    return this.conversations.reduce((res, c) => res + +c.unreadCount, 0);
  }

  @computed
  get unreadGroupCount() {
    return this.groups.reduce((res, c) => res + +c.unreadCount, 0);
  }

  getDefaultConversation(): ConversationInfo | Group {
    if (this.conversations[0]) {
      return this.conversations[0];
    }

    if (this.groups[0]) {
      return this.groups[0];
    }

    return null;
  }

  find(callback): ConversationInfo {
    return this.conversations.find(callback)
      || this.groups.find(callback)
      || this.marketingGroups.find(callback);
  }

  byId(conversationId): ConversationInfo {
    return this.find(c => c.id === conversationId);
  }

  getConversationType(conversationId) {
    const conversation = this.byId(conversationId);
    return conversation ? conversation.type : null;
  }
}