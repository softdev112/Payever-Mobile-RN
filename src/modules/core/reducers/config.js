import { Map } from 'immutable';
import config from '../../../config';

const initialState = Map(config);

/**
 * Share config through store
 */
export default function configReducer(state = initialState) {
  return state;
}