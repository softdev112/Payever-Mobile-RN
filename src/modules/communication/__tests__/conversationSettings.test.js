/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react/native';
import config from '../../../config';
import ConversationSettings from '../screens/ConversationSettings';
import Store from '../../../store';
import MessengerInfo from '../../../store/communication/models/MessengerInfo';
import { messengerData } from '../../../store/communication/__tests__/data';
import navigator from '../../../../__mocks__/navigator';

describe('modules/communication/screens/ConversationSettings', () => {
  let store;
  let communication;

  beforeAll(() => {
    ConversationSettings.prototype.getChildContext = function getContext() {
      return { navigator };
    };

    ConversationSettings.childContextTypes = { navigator: React.PropTypes.object };
  });

  beforeEach(() => {
    store = new Store(config);
    communication = store.communication;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('ConversationSettings should render correctly', () => {
    const conversationId = messengerData.conversations[0].id;
    communication.selectedConversationId = conversationId;
    communication.messengerInfo = new MessengerInfo(messengerData);
    communication.conversations
      .set(conversationId, communication.messengerInfo.byId(conversationId));
    communication.getConversationSettings = jest.fn(
      () => ({ notification: true })
    );

    const tree = renderer.create(
      <Provider
        communication={communication}
      >
        <ConversationSettings navigator={navigator} />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(communication.getConversationSettings).toHaveBeenCalled();
  });
});