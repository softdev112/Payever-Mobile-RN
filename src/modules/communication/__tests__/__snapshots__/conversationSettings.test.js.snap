// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/communication/screens/ConversationSettings ConversationSettings should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderBottomColor": "#e1e1e1",
            "borderBottomWidth": 1,
            "elevation": 8,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "flex-end",
            "marginTop": 15,
            "paddingHorizontal": 18,
            "zIndex": 1000,
          },
          Object {
            "paddingHorizontal": 9,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "left": 18,
            "position": "absolute",
            "top": 0,
            "zIndex": 1002,
          },
          Object {
            "left": 9,
          },
        ]
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="icon-x-24"
          style={
            Array [
              Object {
                "fontFamily": "payeverIcons",
                "fontSize": 24,
              },
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                  Object {
                    "color": "#0084ff",
                  },
                  Array [
                    undefined,
                    Object {
                      "color": "#000",
                    },
                  ],
                ],
              ],
            ]
          }
          touchStyle={
            Object {
              "alignItems": "center",
              "borderColor": "blue",
              "borderWidth": 1,
              "justifyContent": "center",
            }
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0084ff",
                "fontSize": 16,
                "marginLeft": 2,
              },
              Object {
                "fontSize": 16,
                "fontWeight": "400",
                "marginLeft": 2,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1001,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "300",
                "textAlign": "center",
              },
              Object {},
            ]
          }
        >
          Conversation Settings
        </Text>
      </View>
    </View>
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": null,
          "justifyContent": "flex-end",
          "paddingLeft": 20,
          "resizeMode": "stretch",
          "width": null,
          "zIndex": 1002,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      source="icon-alert-32"
      style={
        Array [
          Object {
            "fontFamily": "payeverIcons",
            "fontSize": 32,
          },
          Array [
            Object {
              "fontFamily": "payeverIcons",
              "fontSize": 32,
            },
            Object {
              "color": "#ff3c30",
              "marginBottom": 10,
            },
          ],
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#3d3d3d",
            "fontFamily": "Helvetica Neue",
            "fontWeight": "200",
          },
          Object {
            "fontSize": 15,
            "textAlign": "center",
          },
        ]
      }
    >
      Error while loading settings. Try again later please
    </Text>
  </View>
</View>
`;
